sechub-wrapper-owasp-zap/src/main/java/com/mercedesbenz/sechub/zapwrapper/config/SecHubScanConfigProvider.java

```diff
@@ -8,22 +8,27 @@
 import com.mercedesbenz.sechub.commons.model.SecHubScanConfiguration;
 import com.mercedesbenz.sechub.zapwrapper.cli.ZapWrapperExitCode;
 import com.mercedesbenz.sechub.zapwrapper.cli.ZapWrapperRuntimeException;
+import com.mercedesbenz.sechub.zapwrapper.util.EnvironmentVariableConstants;
+import com.mercedesbenz.sechub.zapwrapper.util.EnvironmentVariableReader;
 
 public class SecHubScanConfigProvider {
-    public SecHubScanConfiguration getSecHubWebConfiguration(File secHubConfigFile) {
-        if (secHubConfigFile == null) {
-            return new SecHubScanConfiguration();
-        }
-        TextFileReader fileReader = new TextFileReader();
+    public SecHubScanConfiguration fetchSecHubScanConfiguration(File secHubConfigFile, EnvironmentVariableReader environmentVariableReader) {
+        if (secHubConfigFile != null) {
+            TextFileReader fileReader = new TextFileReader();
+            try {
+                String sechubConfigJson = fileReader.readTextFromFile(secHubConfigFile);
+                return SecHubScanConfiguration.createFromJSON(sechubConfigJson);
+            } catch (IOException e) {
+                throw new ZapWrapperRuntimeException("Was not able to read sechub config file: " + secHubConfigFile, e, ZapWrapperExitCode.IO_ERROR);
+            }
+        } else if (environmentVariableReader != null) {
```

Comment:
I would change the else if part and do it always...
But you could assert that environmentVariableReader is never null